12/umbraco-cms/reference/routing/custom-middleware.md

```diff
@@ -0,0 +1,70 @@
+---
+description: "Customizing the ASP.NET middleware pipeline in Umbraco"
+---
+
+# Middleware
+
+Middleware is a way of handling requests and responses before or after they are completed. An example can be checking if a user is logged in before adding an item to the cart. In this example it checks something before an http request is made.
+
+In Umbraco we configure some middleware by default and in a specific order based on the [Microsoft Documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-7.0#middleware-order).
+
+You are also able to add your own in-between middleware by using the Umbraco pipeline filters. In addtion you can also add settings such as: the `PrePipeline`, `PostPipeline`, `PreRouting`, `PostRouting`, and [`Endpoints`](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-7.0#endpoints) callbacks.
```

Comment:
```suggestion
You can use Umbraco pipeline filters in case you want to add your own middleware before, in-between or after the default Umbraco middleware. Filters are added by configuring the `UmbracoPipelineOptions` and require an instance of `IUmbracoPipelineFilter` that contains the following callbacks:
```